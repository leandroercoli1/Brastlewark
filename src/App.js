import React from "react";
import Homepage from "containers/homepage";
import { Provider } from "react-redux";
import { store } from "redux/store";
import "styles/global.scss";

function App() {
  return (
    <Provider store={store}>
      <Homepage />
    </Provider>
  );
}

export default App;
