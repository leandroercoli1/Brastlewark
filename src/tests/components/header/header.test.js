import React from "react";
import { fireEvent, screen, render } from "@testing-library/react";
import Header from "components/header";
import App from "App";

describe("renders header correctly", () => {
  it("renders header correctly", () => {
    const {container} = render(<Header/>);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("user sidebar opens on user badge click", async () => {
    render(<App />);
    const userBadge = screen.getByTestId("user-badge");
    expect(userBadge).toBeInTheDocument();
    const userSidebar = screen.getByTestId("user-sidebar");
    expect(userSidebar).toBeInTheDocument();
    expect(userSidebar).toHaveClass("sidebar-slide");
    expect(userSidebar).not.toHaveClass("open");
    fireEvent.click(userBadge);
    expect(userSidebar).toHaveClass("open");
  });
});
