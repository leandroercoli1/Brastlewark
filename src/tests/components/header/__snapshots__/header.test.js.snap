// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly renders header correctly 1`] = `
<div
  class="header"
>
  <div
    class="header-brand"
  >
    <img
      alt="logo"
      class="header-logo"
      src="logo.png"
    />
    <h2>
      Brastlewark
    </h2>
  </div>
  <div
    class="user-badge"
    data-testid="user-badge"
  >
    <img
      alt="logo"
      src="hero.jpg"
    />
  </div>
</div>
`;
